cmake/CMakeLists.txt

```diff
@@ -1057,37 +1060,74 @@ if(PKG_OPT)
 endif()
 
 if(PKG_USER-INTEL)
-    find_package(TBB REQUIRED)
-    find_package(MKL REQUIRED)
+  if(LAMMPS_SIZES STREQUAL BIGBIG)
+    message(FATAL_ERROR "The USER-INTEL Package is not compatible with -DLAMMPS_BIGBIG")
+  endif()
+  add_definitions(-DLMP_USER_INTEL)
 
-    if(NOT CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
-      message(FATAL_ERROR "USER-INTEL is only useful together with intel compiler")
+  set(INTEL_ARCH "cpu" CACHE STRING "Architectures used by USER-INTEL (cpu or knl)")
+  set(INTEL_ARCH_VALUES cpu knl)
+  set_property(CACHE INTEL_ARCH PROPERTY STRINGS ${INTEL_ARCH_VALUES})
+  validate_option(INTEL_ARCH INTEL_ARCH_VALUES)
+  string(TOUPPER ${INTEL_ARCH} INTEL_ARCH)
+
+  find_package(Threads QUIET)
+  if(Threads_FOUND)
+    set(INTEL_LRT_MODE "threads" CACHE STRING "Long-range threads mode (none, threads, or c++11)")
+  else()
+    set(INTEL_LRT_MODE "none" CACHE STRING "Long-range threads mode (none, threads, or c++11)")
+  endif()
+  set(INTEL_LRT_VALUES none threads c++11)
+  set_property(CACHE INTEL_LRT_MODE PROPERTY STRINGS ${INTEL_LRT_VALUES})
+  validate_option(INTEL_LRT_MODE INTEL_LRT_VALUES)
+  string(TOUPPER ${INTEL_LRT_MODE} INTEL_LRT_MODE)
+  if(INTEL_LRT_MODE STREQUAL "THREADS")
+    if(Threads_FOUND)
+      add_definitions(-DLMP_INTEL_USELRT)
+      list(APPEND LAMMPS_LINK_LIBS ${CMAKE_THREAD_LIBS_INIT})
+    else()
+      message(FATAL_ERROR "Must have working threads library for Long-range thread support")
     endif()
+  endif()
+  if(INTEL_LRT_MODE STREQUAL "C++11")
+    add_definitions(-DLMP_INTEL_USERLRT -DLMP_INTEL_LRT11)
+  endif()
 
+  if(CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
     if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 16)
-        message(FATAL_ERROR "USER-INTEL needs at least a 2016 intel compiler, found ${CMAKE_CXX_COMPILER_VERSION}")
+      message(FATAL_ERROR "USER-INTEL needs at least a 2016 Intel compiler, found ${CMAKE_CXX_COMPILER_VERSION}")
     endif()
+  else()
+    message(WARNING "USER-INTEL gives best performance with Intel compilers")
+  endif()
 
-    if(NOT BUILD_OMP)
-        message(FATAL_ERROR "USER-INTEL requires OpenMP")
-    endif()
+  find_package(TBB QUIET)
```

Comment:
Hmm, do we want to do this quietly? The user will have no idea if `TBB` wasn't used. Maybe we should add a note to summary as the end.

Reply:
It isn't that big a deal actually. TBB and MKL are now bundled with the Intel compilers and there is a warning, when configuring USER-INTEL w/o intel compilers. This is still useful for testing and debugging (and there is still some speedup with gcc despite the vectorization directives being ifdef'd out).

It is really only about the malloc library in TBB, that guarantees preserving alignment even for realloc() why this is needed. Since this matters only for using the Intel compiler there is probably best to only warn in that case. when you are not using the intel compiler, it is best to only show people that they are better off using the intel compiler. otherwise they get the wrong impression that using TBB is as important.

Reply:
> TBB and MKL are now bundled with the Intel compilers

That would be the Intel Parallel Studio, but it is possible that during installation, one would have chosen to install only the bare compiler. In that case, it would be nice to have the warning.

Reply:
suitable messages are added (warning for TBB with intel and status message for MKL as the former is more important)